### ISSUE TITLE
CS8619 False positive using ternary operator ?:

### ISSUE THREAD

**Issue by FICTURE7** [dotnet/roslyn#40759]
**Version Used**: 
Visual Studio Community 2019 16.13.5

**Steps to Reproduce**:
1. Use ternary operator with task whose value can be nullable as follows.
```csharp
public Task<string?> ThingAsync(int x)
{
    var what0 = Task.FromResult("");
    var what1 = Task.FromResult(default(string));
    if (x == 0)
        return what0!; // No warning.
    return what1;
}

public Task<string?> ThingTernaryAsync(int x)
{
    var what0 = Task.FromResult("");
    var what1 = Task.FromResult(default(string));
    return x == 0 ? what0! : what1; // Warning.
}
```

Here is the [SharpLab](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQyAMABMugKwDcOOyAzEagQMIEDeOB7BAxAHYCuEEAIYAjCAFMCY7iPFsOBOexpEAbAB5ieAPwA+AgBUAFmG4BzAIIBnAJ7cAxgAoTwAgA8AlIpZf5AN0EATgQA7oaCwIQAvKoAdABiAQD2ALYASmKW/MAOAEQ57hTY8n6BIWHA6ATRyCrxSWkZWQ4wYgBmgk2a7gU+HGCtBA6uVdF4nkXFk8gA7GXheACEhZNKs6Hh6MscAL6UE0q0NRro2npGJqb6YgHSAdZWto7ObuPyrPvF/kHrEVWxCSl0pkINk8j0PiVvuVKtVagCGsDsi12p0Tt0tlNZsNIqMCFo5hEFgQQATNl5dthtkA==) link.

**Expected Behavior**:
No warning in `ThingTernaryAsync(int)`.

**Actual Behavior**:
> Warning	CS8619	Nullability of reference types in value of type 'Task<string>' doesn't match target type 'Task<string?>'.

**Comment by jcouv** [comment]
FYI @cston
